analyzer/consensus/consensus.go

```diff
@@ -343,10 +343,13 @@ func (m *processor) queueTransactionInserts(batch *storage.QueryBatch, data *sto
 			result.Error.Code,
 			message,
 		)
-		batch.Queue(queries.ConsensusAccountNonceUpsert,
-			sender,
-			tx.Nonce+1,
-		)
+		if m.mode != analyzer.FastSyncMode {
+			// Skip during fast sync; will be provided by the genesis.
+			batch.Queue(queries.ConsensusAccountNonceUpsert,
```

Comment:
Hm ... what would that look like? I think we'd need some internal representation of queries, which would be a major overhaul. Otherwise, I don't see a principled way to "go over the batch and [identify and] remove the unneeded updates". I guess we could use the SQL template string as a unique-enough identifier ...

Notwithstanding, I'm not a fan of the idea. It would make it so that when modifying some part of the analyzer (e.g. mint handling, as a random example) and are adding/removing queries there, you'd have to remember to go to an entirely different part of code and add/remove code there also. Conversely, when you're looking at the min-handling code, you wouldn't be able to tell (without looking at a completely separate chunk of code) whether the query that's being added to the batch will _actually_ get executed.

I agree that it doesn't look nice how the `if m.mode` statements pepper the code now, but I think it's more visible in this PR than it will be during actual maintenance of the indexer. Future feature work and debugging will tend to focus on one aspect of the blockchain at a time, like the mints above.

Reply:
Yeah, likely no nice way to achieve this in `go` regardless.

>Notwithstanding, I'm not a fan of the idea. It would make it so that when modifying some part of the analyzer (e.g. mint handling, as a random example) and are adding/removing queries there, you'd have to remember to go to an entirely different part of code and add/remove code there also. Conversely, when you're looking at the min-handling code, you wouldn't be able to tell (without looking at a completely separate chunk of code) whether the query that's being added to the batch will actually get executed.

Yeah, I imagined that info being part of the SQL query definition (since this is Go the nicest I could think of was maybe an SQL comment with a special tag), so you wouldn't need to jump to/be aware of a separate chunk of code, as the tag could be quite self-explanatory. Basically, it would just move the current go code that does: `"if fast-sync" skip query` into the query definition itself.

One benefit is that the code path for fast-sync and slow-sync would be more alike. It would be harder to introduce bugs that would only affect one of the two paths (since it would be ensured that the only difference is execution of some queries). While currently, (in theory), one of the "if" switches could also skip some side-effect incurring code.

However, I'm not convinced that this minor benefit outweighs the added complexity, so am completely fine with keeping the code as it is.

> but I think it's more visible in this PR than it will be during actual maintenance of the indexer

Probably true :+1: